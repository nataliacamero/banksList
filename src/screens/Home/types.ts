import {RouteProp} from '@react-navigation/native';
import {NativeStackNavigationProp} from '@react-navigation/native-stack';
import {StackParamList} from '../../../navigation/types';

export interface IResponse {
  age: number;
  bankName: string;
  description: string;
  url: string;
  id: string;
}

export type ImageDimensions = {
  width: number;
  height: number;
};

export type HomeScreenNavigationProp = NativeStackNavigationProp<
  StackParamList,
  'ProductsList'
>;

export type HomeRouteProp = RouteProp<StackParamList, 'Home'>;
