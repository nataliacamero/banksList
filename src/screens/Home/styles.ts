import {SafeAreaView, Text, TouchableOpacity, View} from 'react-native';
import styled from 'styled-components';

export const SafeAreaStyled = styled(SafeAreaView)`
  background-color: white;
`;

export const Container = styled(View)`
  display: flex;
  flex-flow: column nowrap;
  justify-content: center;
  align-items: center;
  width: 100%;
  padding-top: 50px;
  padding-bottom: 50px;
`;

export const BoldParagraph = styled(Text)`
  font-size: 16px;
  font-weight: 700;
  color: #020202;
`;
