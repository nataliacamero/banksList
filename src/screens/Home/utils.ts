import {ImageDimensions} from './types';

export const customeDimensionsLogo = (itemName: string): ImageDimensions => {
  const customeHeight = itemName === 'Banregio' ? 80 : 120;
  return {
    width: 160,
    height: customeHeight,
  };
};
