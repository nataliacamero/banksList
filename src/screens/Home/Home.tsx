import {useNavigation} from '@react-navigation/native';
import React, {useEffect, useState} from 'react';
import {FlatList, Image, StatusBar, Text} from 'react-native';
import {Header} from '../../components';
import {BoldParagraph, Container, SafeAreaStyled} from './styles';
import {HomeScreenNavigationProp} from './types';
import {generateId} from '../../utils/functions';
import getBanksInfo from '../../services/getBanksInfo';

interface IResponse {
  age: number;
  bankName: string;
  description: string;
  url: string;
  id: string;
}

type ImageDimensions = {
  width: number;
  height: number;
};

const dimensionsImages = (itemName: string): ImageDimensions => {
  const customeHeight = itemName === 'Banregio' ? 80 : 120;
  return {
    width: 160,
    height: customeHeight,
  };
};

const Home: () => JSX.Element = () => {
  const navigation = useNavigation<HomeScreenNavigationProp>();
  const [data, setData] = useState<IResponse[] | null>(null);

  useEffect(() => {
    if (!data) {
      getBanksInfo()
        .then(dataService => {
          let dataWithIdProperty =
            dataService &&
            dataService?.map((item: IResponse) => {
              item.id = generateId();
              console.log(item.url);

              return item;
            });
          setData(dataWithIdProperty);
        })
        .catch((error: {message: string}) => console.error(error.message));
    }
  }, [data]);

  return (
    <SafeAreaStyled>
      <StatusBar barStyle={'dark-content'} />
      <Header
        title={'Available Banks'}
        subtitle={'Find your preferred bank in our list of available banks.'}
        testID="header-products-list"
      />
      <Container testID={`container-sku-product`}>
        {data && data ? (
          <FlatList<IResponse>
            data={data}
            renderItem={({item}) => (
              <Image
                // eslint-disable-next-line react-native/no-inline-styles
                style={dimensionsImages(item.bankName)}
                source={{
                  uri: `${item.url}`,
                }}
              />
            )}
          />
        ) : (
          <BoldParagraph>Loading...</BoldParagraph>
        )}
      </Container>
    </SafeAreaStyled>
  );
};

export default Home;
