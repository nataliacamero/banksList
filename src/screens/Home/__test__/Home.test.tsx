import React from 'react';
import {cleanup, render, screen} from '@testing-library/react-native';
import {useNavigation} from '@react-navigation/native';
import Home from '../Home';

jest.useFakeTimers();
jest.mock('@react-navigation/native', () => ({
  useRoute: jest.fn(),
  useNavigation: jest.fn(),
}));

const useNavigationMock = useNavigation as jest.MockedFunction<
  typeof useNavigation
>;

afterEach(cleanup);

describe('<Home/>', () => {
  beforeEach(() => {
    useNavigationMock.mockReset();
  });

  it('Should render the component', () => {
    const rendered = render(<Home />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('Should render the Title -Available Banks-', () => {
    render(<Home />);
    const {getByTestId} = screen;
    const title = getByTestId('title-header-home');
    expect(title.children[0]).toBe('Available Banks');
  });
  it('Should render the subtitle -Here you can...-', () => {
    render(<Home />);
    const {getByTestId} = screen;
    const subtitle = getByTestId('subtitle-header-home');
    expect(subtitle.children[0]).toBe(
      'Find your preferred bank in our list of available banks.',
    );
  });
});
