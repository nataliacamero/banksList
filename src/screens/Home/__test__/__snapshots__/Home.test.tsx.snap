// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> Should render the component 1`] = `
<RCTSafeAreaView
  className="sc-eDDNvR hDMTaq"
>
  <View
    className="sc-beqWaB bCUvjq"
  >
    <Text
      className="sc-gueYoa VqWBx"
      testID="title-header-home"
    >
      Available Banks
    </Text>
    <Text
      className="sc-dmqHEX cobd"
      testID="subtitle-header-home"
    >
      Find your preferred bank in our list of available banks.
    </Text>
  </View>
  <View
    className="sc-jTrPJq hwdmtr"
    testID="container-sku-product"
  >
    <RCTScrollView
      data={[]}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      testID="banks-list"
      viewabilityConfigCallbackPairs={[]}
    >
      <View />
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
